### PR TITLE
Use strict mode typescript

### PR DESCRIPTION
**Pull request by AAJELLAL**
**What kind of change does this PR introduce?**

Activate Strict mode in TypeScript and fix code errors related to it.

**What is the current behavior?**

Typescript is used  with strict mode disabled.

**What is the new behavior (if this is a feature change)?**
Typescript  with strict mode enabled.

### CODE REVIEW

**Review by archendel — COMMENTED** [review]
try to add other configs as well

File: src/single-line-diagram-viewer.ts [hunk 1]
```diff
@@ -696,38 +699,41 @@ export class SingleLineDiagramViewer {
...
-
-            svgBus.addEventListener('contextmenu', (event) => {
+        buses?.forEach((bus) => {
+            const svgBus: HTMLElement | null = this.container?.querySelector(
+                '#' + bus.id
+            );
+            if (svgBus) {
+                svgBus.style.cursor = 'pointer';
+            }
+            svgBus?.addEventListener('contextmenu', (event) => {
```
Comment (etiennehomer) [inline src/single-line-diagram-viewer.ts:736]: Maybe move svgBus.addEventListener in the if above and you can remove the '?'

Comment (AAJELLAL) [reply src/single-line-diagram-viewer.ts]: Indeed. It's done

File: src/single-line-diagram-viewer.ts [hunk 2]
```diff
@@ -507,11 +507,12 @@ export class SingleLineDiagramViewer {
...
                     return;
                 }
-                const meta = svgMetadata.nodes.find(
+                const meta = svgMetadata?.nodes.find(
                     (other) => other.id === element.id
                 );
                 if (meta !== undefined && meta !== null) {
-                    this.onNextVoltageCallback(meta.nextVId);
+                    this.onNextVoltageCallback &&
+                        this.onNextVoltageCallback(meta.nextVId);
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/single-line-diagram-viewer.ts:515]: Personally I never used this syntax but why not.

**Comment by AAJELLAL** [comment]
> try to add other configs as well

Done, but some of the properties can't be used such as  "allowJs" and   "isolatedModules".

**Review by etiennehomer — DISMISSED** [review]

**Review by archendel — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**